src/test/scala/org/constellation/invertedmap/InvertedMapTest.scala

```diff
@@ -53,10 +53,8 @@ class InvertedMapTest extends AnyFreeSpec with Matchers {
     (m + (("node4", allReady))).get("node4").isDefined shouldBe true
   }
 
-  "does not allow to set same key again" in {
-    assertThrows[AssertionError] {
-      m + (("node3", allReady))
-    }
+  "does allow to set same key again" in {
+    m + (("node3", allReady))
```

Comment:
the same key cannot be set again if it exists in the inverted map. you will have a duplication then which is a forbidden state

Reply:
to set the key again, you have to either remove it first and then add it again with a different value, or just use an update that does that under the hood

Reply:
as an example, you cannot have an inverted map like that:
```
all ready -> node3
one offline -> node3
```
because it's ambiguous when you invert it:
```
node3 -> ???
```

Reply:
hence the `key` must exist once only

Reply:
You're probably right about this. I will look at it tomorrow.

Reply:
Ok, so two things:
1. In you're example the `InvertedMap` can't have repeating values, which is not the case for this implementation. The `InvertedMap[A, B]` doesn't have to be injective (różnowartościowa), because the underlying map is of type `Map[B, Set[A]]` which solves the issue.
2. This plus method is just an alias to the `updated` method, which first removes the existing association and then adds a new one, so the underlying data is kept consistent.

You may wonder why I'm aliasing this function at all and why can't I just change the invocations of `+` to `updated`. So the reason is that  we're not calling it directly in our code, but rather indirectly by calling the `MapLike`'s `++` (add all) method, which in turn calls `+` (add) method.

Reply:
I agree with Marcin. `Set` does not solve the problem. `Map[B, Set[A]]` only makes `B` unique and `A` unique *wthin the single set* but we want `A` to be unique globally in the whole map:

```
all ready -> Set(node3, ...)
one offline -> Set(node3, ...)
```

Such state is incorrect because it says that "node3 has all the nodes ready" and "node3 has one node offline".